src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs

```diff
@@ -1400,6 +1405,11 @@ private void EnterParameters(ImmutableArray<ParameterSymbol> parameters)
 
         protected virtual void EnterParameter(ParameterSymbol parameter)
         {
+            if ((parameter.ContainingSymbol as MethodSymbol)?.MethodKind == MethodKind.LocalFunction)
```

Comment:
(As I said below, attributes are not allowed on local functions - putting this code in `VisitLocalFunctionStatement` would duplicate logic that could become out-of-sync, so I'd prefer not to do that. Let me know if this comment is still applicable.)